tensorflow-frameworks/tensorflow-data/src/main/java/org/tensorflow/data/Utils.java

```diff
@@ -0,0 +1,34 @@
+package org.tensorflow.data;
+
+import org.tensorflow.tools.Shape;
+
+public class Utils {
+
+    private static Shape head(Shape shape) {
+        return Shape.of(shape.size(0));
+    }
+
+    public static Shape tail(Shape shape) {
+        long[] tail = new long[shape.numDimensions() - 1];
+        for (int i = 1; i < shape.numDimensions(); i++) {
+            tail[i - 1] = shape.size(i);
+        }
+
+        return Shape.of(tail);
+    }
+
+    public static long[] shapeArray(Shape shape) {
+        long[] arr = new long[shape.numDimensions()];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = shape.size(i);
+        }
+        return arr;

```

Comment:
that could be replaced simply by `shape.asArray()`, unless you need a copy, which `Arrays.copyOf` can provide you